### PR TITLE
chore(repo): Clean up users in e2e tests

### PR DESCRIPTION
**Pull request by dnys1**
- Adds teardown calls to Authenticator e2e tests. Should resolve some flakiness with phone number tests.
- Adds a scheduled cron job to the e2e infra to delete all users every day.

### CODE REVIEW

**Review by NikaHsn — APPROVED** [review]

File: infra/lib/common.ts [hunk 1]
```diff
@@ -136,6 +140,30 @@ export abstract class IntegrationTestStackEnvironment<
...
+      environment: {
+        USER_POOL_ID: userPool.userPoolId,
+      },
+      timeout: cdk.Duration.minutes(5),
+    });
+    userPool.grant(lambda, "cognito-idp:ListUsers", "cognito-idp:AdminDeleteUser");
+
+    const cronRule = new events.Rule(this, 'cleanup-users-rule', {
+      // Run daily at midnight
+      schedule: events.Schedule.expression('cron(0 0 ? * MON-FRI *)'),
```
Comment (ragingsquirrel3) [inline infra/lib/common.ts:163]: nit: Could this run every day, not just mon-fri? In the future, am thinking of scheduling some very long running, high-dimension matrix suites on weekends to limit pool load, and might as well keep this running. IIUC, the lambda will run fast/bill low when pool is empty.

**Review by ragingsquirrel3 — APPROVED** [review]
approved w a few non-blocking comments

File: infra/lib/cleanup-users.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export const handler: lambda.Handler = async (event) => {
+    console.log(`Got event: ${JSON.stringify(event, null, 2)}`);
+
+    const users: UserType[] = [];
+    try {
+        console.log('Listing users...');
+
+        let nextToken: string | undefined;
+        do {
+            const usersResp = await CLIENT.listUsers({
```
[the remark this answers is not in the document]
Comment (dnys1) [reply infra/lib/cleanup-users.ts:19]: That's why I scheduled it to run at midnight. I think keeping the logic simple here is sufficient.

File: infra/lib/common.ts (hunk 1 above)
Comment (dnys1) [reply infra/lib/common.ts:163]: Can we revisit then if it becomes an issue?

Comment (ragingsquirrel3) [reply infra/lib/common.ts:163]: Sure, that's fine. We already run the integ tests daily including weekends but this should be fine for now.

**Pull request merged**